client/wfm/src/result/StepResult.ts

```diff
@@ -0,0 +1,35 @@
+import { STATUS } from '../status';
+import { Step } from '../step/Step';
+
+/**
+ * Output of a single {@link Step} in the context of a {@link WorkOrder}
+ */
+export interface StepResult {
+  /**
+   * Form data output of the {@link Step}'s execution
+   */
+  submission?: {
```

Comment:
This format is really strange and was added to reference submissions by type. In practice this may lead to multiple issues (for example when same key is used twice). I think we may pull out key into object to be more object friendly:
```{
key:string, 
data:any
}```